lib/http.ex

```diff
@@ -40,10 +40,11 @@ defmodule Gitea.Http do
   """
   @spec parse_body_response({atom, String.t()} | {:error, any}) :: {:ok, map} | {:error, any}
   def parse_body_response({:error, err}), do: {:error, err}
+  def parse_body_response({:ok, response = %{status_code: 204}}), do: {:ok, response}
```

Comment:
Check for status code here instead of checking if the body exists as a 204 status code response has an empty body.
see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204 and https://gitea-server.fly.dev/api/swagger#/organization/orgDelete

Reply:
Worth including this comment in-line (in the code) as it will be lost here in the PR ...

Reply:
The reason I've added this definition is to avoid changing the code below:
```elixir
 if body == nil || byte_size(body) == 0 do
```
My first thought was to remove the `byte_size(body) == 0` condition as I don't think `parse_body_response` should return `{:error, :no_body}` when the body is an empty string (it is defined like this when deleting orgs or repos). However I didn't want to break existing tests in case I missed something.

I'll add a comment in the code